plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -362,17 +378,80 @@ func (m *MQTTConsumer) createOpts() (*mqtt.ClientOptions, error) {
 				server = "ssl://" + server
 			}
 		}
-
 		opts.AddBroker(server)
 	}
 	opts.SetAutoReconnect(false)
 	opts.SetKeepAlive(time.Second * 60)
 	opts.SetCleanSession(!m.PersistentSession)
 	opts.SetConnectionLostHandler(m.onConnectionLost)
-
 	return opts, nil
 }
 
+// parseMeasurement gets a single value from the incoming topic based on the user configuration (TopicParsing.Measurement)
+func parseMeasurement(keys []string, values []string) (string, error) {
+	for i, k := range keys {
+		if k != "_" {
+			return values[i], nil
+		}
+	}
+	return "", fmt.Errorf("no measurements found")
+}
+
+// parseTags gets multiple tags from the topic based on the user configuration (TopicParsing.Tags)
+func parseTags(keys []string, values []string) (map[string]string, error) {
+	results := make(map[string]string)
+	for i, k := range keys {
+		if k != "_" {
+			results[k] = values[i]
+		}
+	}
+	if len(results) == 0 {
+		return nil, fmt.Errorf("no tags found")
+	}
```

Comment:
This will trigger in case someone configured `tags = "/_/_/_"` for whatever reason... :-) Is this intended?

Reply:
yes, it will error out as i believe it is an invalid config

Reply:
Tags aren't required for valid Line Protocol but if someone doesn't have any Tags, they shouldn't be parsing any out and should therefore not be using the `tags` field of this parsing feature.  That said, it's possible someone might do that if they think the field is required?

Reply:
This is intended as it is an invalid config.